compiler/parser/statement_parsing.go

```diff
@@ -137,25 +114,70 @@ func (p *Parser) parseParameters() []ast.Expression {
 
 	for p.peekTokenIs(token.Comma) {
 		p.nextToken()
+		p.nextToken()
 
-		if p.peekTokenIs(token.Asterisk) {
+		if p.curTokenIs(token.Asterisk) && !p.peekTokenIs(token.Ident) {
+			p.expectPeek(token.Ident)
 			break
 		}
 
-		p.nextToken()
-
 		param := p.parseExpression(NORMAL)
-
-		if paramDuplicated(params, param) {
-			p.error = &Error{Message: fmt.Sprintf("Duplicate argument name: \"%s\". Line: %d", getArgName(param), p.curToken.Line), errType: SyntaxError}
-		}
 		params = append(params, param)
 	}
 
+	p.checkMethodParameters(params)
+
 	p.fsm.Event(backToNormal)
 	return params
 }
 
+func (p *Parser) checkMethodParameters(params []ast.Expression) {
```

Comment:
@shuheiktgw I added a function for checking parameters, currently only checking their order and duplication.